src/modules/esp32/PaxcounterModule.cpp

```diff
@@ -76,12 +97,14 @@ int32_t PaxcounterModule::runOnce()
             libpax_update_config(&configuration);
 
             // internal processing initialization
-            libpax_counter_init(NullFunc, &count_from_libpax, UINT16_MAX, 1);
+            libpax_counter_init(handlePaxCounterReportRequest, &count_from_libpax,
+                                moduleConfig.paxcounter.paxcounter_update_interval, 0);
             libpax_counter_start();
         } else {
             sendInfo(NODENUM_BROADCAST);
         }
-        return getConfiguredOrDefaultMs(moduleConfig.paxcounter.paxcounter_update_interval, default_broadcast_interval_secs);
+        // we check every second if the counter had new data to send
+        return 1000;
```

Comment:
this is rather heavy on the threading system, like polling for a flag. My suggestion is to keep the original interval (no change here) and trigger the osthread from the outside when new data arrived. See comment above